IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -250,6 +275,21 @@ - (void)saveTelemetryToDisk
     {
         NSData *dataToArchive = [NSKeyedArchiver msidArchivedDataWithRootObject:self requiringSecureCoding:YES error:nil];
         
+        // If data is larger than 10x the 4kB limit, start removing errors from the telemetry
+        // object until it fits under the limit. Limit is 10x to allow for additional metadata
+        // that NSKeyedArchiver adds to the archive
+        if (dataToArchive.length > [MSIDLastRequestTelemetry telemetryStringSizeLimit] * 10)
+        {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, nil, @"Telemetry size over limit when saving to disk, cutting down to limit", nil);
+            
+            while (_errorsInfo.count > 0 &&
```

Comment:
If anyone has any suggestions or ideas for a more elegant way to continuously check the size of the object as it changes without doing so many operations, I'm be interested to hear it.

An idea that came up was to measure the size of an abstract object (like a serialization of the errorsInfo) that would scale in size the same way that the object itself would. Would creating a different serialization function just for this be worth doing?

Reply:
+1, let's discuss alternative options to optimize :)